### DISCUSSION TITLE
Modification - Checkbox size variant

### DISCUSSION THREAD

**Discussion by Devaney** [Legal-and-General/canopy#874] [Canopy requests]
Can we create a variant for the checkbox to allow larger sizes?
Would this be added as a `ToggleVarient` or as a new ToggleSize `@Input() size: ToggleSize;`?

The design below show the checkbox with a width of 24px and a height of 24px.
![image](https://user-images.githubusercontent.com/7091410/169793618-2e8e1c28-1cd2-401d-ba62-7775c3ee1ceb.png)

**Comment by elenagarrone** [comment]
I've seen these larger checkboxes in other designs and I've asked why do we need two different sizes but can't remember the answer 🤔  to me it will just seem inconsistent to have them of different sizes in the same application - do we know of other libraries with this functionality? Can we make them all larger?
 
@alexcanning I'm assuming you are the best to answer these questions?

From a technical point of view, if we go with two different sizes, I would probably have it `@Input() size: ToggleSize;`.

**Reply by alexcanning** [reply]
We had originally conceived of having 2 sizes depending on purpose. The current size being used in self service scenarios, and a larger size being used on more content rich pages.

The current size feels a bit small, even for self-service purposes. The solution might be to increase the size of the current checkbox rather than create a new variant.

Another consideration is that a criticism of Canopy is that the components are too large to be used on functionally-dense applications, so it might be worth keeping the smaller size around for these purposes in future.

**Reply by alexcanning** [reply]
Also, should consider modifying the radio button at the same time.

**Reply by elenagarrone** [reply]
Ok, I guess we can go with two sizes then? `sm` and `lg`

**Reply by elenagarrone** [reply]
Agreed on adding `sm` and `lg` sizes to both the `radio` and `checkbox` inputs.

This is currently being worked on.

**Comment by elenagarrone** [comment]
PR: https://github.com/Legal-and-General/canopy/pull/983

**Comment by elenagarrone** [comment]
Available in release [8.1.0](https://github.com/Legal-and-General/canopy/releases/tag/v8.1.0).
